python/simulators/CMakeLists.txt

```diff
@@ -71,3 +60,32 @@ if(OPM_ENABLE_PYTHON_TESTS)
       -E env PYTHONPATH=${PYTHON_PATH} ${PYTHON_EXECUTABLE}
       -m unittest test/test_schedule.py)
 endif()
+
+set(_opm_common_python_script_dir "${opm-common_DIR}/python")
+if (NOT EXISTS "${_opm_common_python_script_dir}/install.py" )
+  # if the install.py script does not exist, assume that we are building against an installed opm-common
+  set(_opm_common_python_script_dir "${opm-common_PYTHON_COMMON_DIR}")
+endif()
+
+# NOTE: instead of using file( COPY ...) which copies the files at configure time (not at build time)
+#  we should add copying of the files at build time such that running "make" or "make all" will
+#  update the files if they have been modified. We use the install.py script in opm-common, see also
+#  CMakeLists.txt in opm-common
+add_custom_target(copy_python ALL
+    COMMAND ${PYTHON_EXECUTABLE} "${_opm_common_python_script_dir}/install.py"
+      ${PROJECT_SOURCE_DIR}/python ${PROJECT_BINARY_DIR} 0)
+
+# Since the installation of Python code is nonstandard it is protected by an
+# extra cmake switch, OPM_INSTALL_PYTHON. If you prefer you can still invoke
+# setup.py install manually - optionally with the generated script
+# setup-install.sh - and completely bypass cmake in the installation phase.
+if (OPM_INSTALL_PYTHON)
+  include(PyInstallPrefix)  # from opm-common
+  install(TARGETS simulators DESTINATION ${DEST_PREFIX}${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_PREFIX}/opm)
+  install(
+    CODE "execute_process(
+      COMMAND ${PYTHON_EXECUTABLE}
+              ${_opm_common_python_script_dir}/install.py
+              ${PROJECT_BINARY_DIR}/python/opm2
+              ${DEST_PREFIX}${CMAKE_INSTALL_PREFIX}/${PYTHON_INSTALL_PREFIX} 1)")
```

Comment:
Not a big fan of not using CMake (might have some strange implications if people try to reset some variables when calling make. What is the rational behind this and how was this done before?

Reply:
opm2 will vanish again, right?

Reply:
I suggest we use the `install.py` until with run into an explicit problem with it. To replace it with cmake, would require listing all the files explicitly in the `add_custom_target(copy_python ALL)`  with entries like `COMMAND ${CMAKE_COMMAND} -E copy_if_different ${PROJECT_SOURCE_DIR}/python/some/dir/file1 ${PROJECT_BINARY_DIR}/python/some/dir` for each file, right?

>opm2 will vanish again, right?

@blattms Yes,  #3933 should remove `opm2` and replace it with `opm`